src/Compilers/Core/Portable/CodeGen/PrivateImplementationDetails.cs

```diff
@@ -292,8 +319,131 @@ internal Cci.IFieldReference CreateDataField(ImmutableArray<byte> data, ushort a
                     _ => hex
                 };
 
-                return new MappedField(name, this, type, key.Data);
-            });
+                return new MappedField(name, @this, type, data);
+            },
+            this);
+        }
+
+        internal DataSectionStringType? GetOrCreateDataSectionStringLiteralType(
```

Comment:
>DataSectionStringType? [](http://example.com/codeflow?start=17&length=22)

Does this function ever return `null` value? The name of the function implies that it doesn't.